### ISSUE TITLE
What is the bare minimum requirements to run Codecov with a .NET Core 2.2.8 application?

### ISSUE THREAD

**Issue by PureKrome** [coverlet-coverage/coverlet#1078]
We have some "legacy" .NET Core 2.2.8/2.2.207 applications. We're trying to add codecov before we can migrate up to 3.1.x or 5.x.

When we try and follow the info set out in the coverlet docs/readme, etc ... we get the following error:

> Starting test execution, please wait...
> Data collector 'XPlat code coverage' message: [coverlet]System.MissingMethodException: Make sure to use .NET core SDK Version >= 2.2.300 ---> System.MissingMethodException: Method not found: '!!0 Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection.SessionStartEventArgs.GetPropertyValue(System.String)'.
   at Coverlet.Collector.DataCollection.CoverletCoverageCollector.GetPropertyValueWrapper(SessionStartEventArgs sessionStartEventArgs)
   at Coverlet.Collector.DataCollection.CoverletCoverageCollector.GetTestModules(SessionStartEventArgs sessionStartEventArgs) in /_/src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs:line 194
   --- End of inner exception stack trace ---
   at Coverlet.Collector.DataCollection.CoverletCoverageCollector.GetTestModules(SessionStartEventArgs sessionStartEventArgs) in /_/src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs:line 205
   at Coverlet.Collector.DataCollection.CoverletCoverageCollector.OnSessionStart(Object sender, SessionStartEventArgs sessionStartEventArgs) in /_/src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs:line 126.

Here's a 100% simple repo:

```
docker pull mcr.microsoft.com/dotnet/core/sdk:2.2.207
docker run --rm -it mcr.microsoft.com/dotnet/core/sdk:2.2.207 /bin/bash
mkdir app
cd app
dotnet new xunit
dotnet add package coverlet.collector --version 1.3.0
dotnet add package microsoft.net.test.sdk
dotnet test --collect:"XPlat Code Coverage"
```

so we
- start a docker instance of the .NET Core 2.2.207 SDK
- create a new xunit project
- add coverlet.collector
- updated the Test sdk.
- test!

so can someone please help / what are we doing wrong?

SIDE NOTE: we understand that 2.2 is not LTS and we're just not in a position just yet to move the sdk to 2.1 or 3.1 ... so we're hoping it's _easier/quicker_ to just get our codecov to work (against the existing codebase) instead of doing and SDK migration (which _should_ be an easy thing .. but we've got processes to follow)

**Comment by MarcoRossignoli** [comment]
This is the minimum requirement for collectors integration https://github.com/coverlet-coverage/coverlet#requirements

**Comment by PureKrome** [comment]
> This is the minimum requirement for collectors integration https://github.com/coverlet-coverage/coverlet#requirements you should try to upgrade sdk to `2.2.401`

and
> You need to be running .NET Core SDK v2.2.401 or newer

[if you check out the downloads](https://dotnet.microsoft.com/download/dotnet-core/2.2) page, 2.2.207 is the latest, though?

and 401 is 4 months older?

> You need to reference version 16.5.0 and above of Microsoft.NET.Test.Sdk

yep, that demo is the most recent

**Comment by MarcoRossignoli** [comment]
Oh I didn't noticed that release dates, anyway I did a test and I confirm that with `2.2.207` it fails but with `2.2.401` and `2.2.300` it works. So I confirm that it's strange also to me 🤔 .

I think it's better open an issue here https://github.com/microsoft/vstest/issues we don't own nothing about SDK release and the first collectors version was done by vstest MS guys.
After that we only maintained/fixed collectors related issue. But we're not in charge to deploy the SDK integraton.

**Comment by PureKrome** [comment]
Thanks so much @MarcoRossignoli for this! I totally didn't think of trying the other releases ... feels like that was so obvious.

Looking forward to seeing what VSTest team have to say.

**Comment by petli** [comment]
The first digit in what looks like the patch version of SDKs is really a feature level: https://docs.microsoft.com/en-us/dotnet/core/versions/

So 2.2.402 means patch version 2 of feature level 4 of .NET Core 2.2, and 2.2.207 is feature level 2 patch version 7. For your restricted update options it may still be possible to go to a higher feature level but still stay on 2.2, and thus use 2.2.402?

**Comment by PureKrome** [comment]
Hi @petli - sorry, I don't understand what you're saying.

Are you saying that version 402 is actually 'newer/higher' than 207 ... despite the fact that 207 was released AFTER 402?

Which means what? people should be using the higher _versions_ .. and not the most _recent_ version (which makes things a bit misleading on the .NET downloads page).

**Comment by petli** [comment]
Think of it this way: these are version 2.2.4.02 versus 2.2.2.07.  You are currently on 2.2.2.07, so you'd need to upgrade to 2.2.3.xx or 2.2.4.xx to be able to use coverlet, and might then go to the latest patch release of the highest feature level, i.e. 2.2.402.  

207 might have been released after 402 to fix some bug that only affects feature level 2.2.2xx, hence being later.

**Comment by PureKrome** [comment]
> 207 might have been released after 402 to fix some bug that only affects feature level 2.2.2xx, hence being later.

Okay! so if 402 > 207 .. then I wish they would list the releases, as such.

If _feels like_ the most recently release == the most recent/highest version.

but what you are saying is that the highest number (regardless of release date) is the 'highest' version. So it's definitely following semver, then.

For example -> if someone was told to go and download the 'most recent' version of 2.2.* .. or update their 2.2* to the 'most recent' .. by looking at the official downloads page, it would _look like_ 207 is the most recent version people should use. not 402.

So if 402 > 207, then yep .. code is good, etc.

**Comment by petli** [comment]
I don't know what those product versions are, I've only had reason to study the SDK versions. Anyway, can you use 2.2.402 and does that solve your issue?

**Comment by PureKrome** [comment]
> Anyway, can you use 2.2.402 and does that solve your issue?

Yep - that should work if the 402 > 207. :)

**Comment by MarcoRossignoli** [comment]
2.2 is out of support so VSTest team won't fix this mismatch, also this SDK is out of support https://dotnet.microsoft.com/platform/support/policy/dotnet-core. I'm going to close.
